parmed/parameters.py

```diff
@@ -217,6 +217,8 @@ def from_structure(cls, struct, allow_unequal_duplicates=True):
                     other_atoms = list(atom.atom_type.nbfix.keys())
                     for other_atom in other_atoms:
                         (rmin, epsilon, rmin14, epsilon14) = atom.atom_type.nbfix[other_atom]
+                        if (other_atom, atom.type) in params.nbfix_types.keys():
```

Comment:
Didn't know it would go through the keys by default, thanks!

And just because I'm curious - why would it be slower to use `.keys()`? This does not seem intuitive to me.

Reply:
> And just because I'm curious - why would it be slower to use `.keys()`? This does not seem intuitive to me.

The effect is modest.  `.keys()` adds a few steps compared to omitting it:

* Calling `__getattr__` on the dict instance
* Instantiating a `dict_keys` object and wrapping the dict keys in a view
* Calls `__contains__` on the created `dict_keys` object

Each step is O(1) (although the prefactor on the last step is the largest since it has to compute a hash).

By contrast, omitting `.keys()` just:

* Calls `__contains__` on the dict object

Here's a quick timing from Jupyter console:

```python
In [2]: big_dict = dict(zip(range(10000), range(10000)))

In [3]: import random

In [4]: bigrange = list(range(10000))

In [5]: %timeit random.choice(bigrange) in big_dict
886 ns ± 11.6 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [6]: %timeit random.choice(bigrange) in big_dict.keys()
978 ns ± 7.63 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

So on my (slow) computer, `(other_atom, atom.type) in params.nbfix_types` would be about 10% faster than `(other_atom, atom.type) in params.nbfix_types.keys()`

Reply:
Good to know, thanks!